clowdapp.yaml

```diff
@@ -111,9 +111,8 @@ objects:
           - bash
           - -c
           - |
-            for i in {1..120}; do sleep 1;
-            if bundle exec rake --trace db:abort_if_pending_migrations;
-            then exit 0; fi; done; exit 1
+            set -e
+            RAILS_ENV="${RAILS_ENV}" bundle exec rake --trace db:debug db:migrate
```

Comment:
We might need to extract this to a single pod just for migrations (like cyndi-init). I am not sure if the concurrency will work if it spins up more than one pod at the same time.

Reply:
It seemed to work fine when deploying in the ephemeral environment yesterday. I'm confused by your statement, though:

> I am not sure if the concurrency will work if it spins up more than one pod at the same time.

What concurrency? How does this change the number of pods started simultaneously? It seems the wait_for_migrations check works great to keep other services waiting in their init containers.

Reply:
The `wait_for_migrations` is run on non-backend pods (consumer, sidekiq...). Here the `db:migrate` is run on any backend pod. If there would be a deployment, there would be at least two pods started. So the question is more on what would happen if `db:migrate` is run simultaneously more than once. Database would guarantee concurrency to some level with transactions, but I'm not sure how it would behave within Rails.  Whether the second pod would reapply the same migration, after a lock is lifted, and fail or not.

Reply:
Ah! Yes, okay. I see now. Thank you for the explanation :) I guess we can add this add a job and include it in the CJI.

Reply:
Ok, updated. Nice catch, btw! That probably wouldn't have worked very well :laughing: Let me know if you see issues with using a job like this.

Reply:
One-time job with a finite number of retries will do I guess. :+1: